### PR TITLE
AMDGPU: Add gfx950 subtarget definitions

### PR DESCRIPTION
**Pull request by arsenm**
Mostly a stub, but adds some baseline tests and
tests for removed instructions.

### CODE REVIEW

**Comment by arsenm** [comment]
* **#116309** <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/116309?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#116308** <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/116308?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#116307** <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/116307?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/116307?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

**Review by shiltian — APPROVED** [review]

File: llvm/tools/llvm-readobj/ELFDumper.cpp [hunk 1]
```diff
@@ -1617,6 +1617,7 @@ const EnumEntry<unsigned> ElfHeaderMipsFlags[] = {
   ENUM_ENT(EF_AMDGPU_MACH_AMDGCN_GFX90A, "gfx90a"),                            \
   ENUM_ENT(EF_AMDGPU_MACH_AMDGCN_GFX90C, "gfx90c"),                            \
   ENUM_ENT(EF_AMDGPU_MACH_AMDGCN_GFX940, "gfx940"),                            \
+  ENUM_ENT(EF_AMDGPU_MACH_AMDGCN_GFX950, "gfx950"),                            \
```
Comment (jayfoad) [inline llvm/tools/llvm-readobj/ELFDumper.cpp:1620]: Move this down after gfx942

File: llvm/test/tools/llvm-objdump/ELF/AMDGPU/subtarget.ll [hunk 2]
```diff
@@ -146,6 +146,11 @@ define amdgpu_kernel void @test_kernel() {
 ; RUN: llc -mtriple=amdgcn-amd-amdhsa --amdhsa-code-object-version=6 -mcpu=gfx9-generic -filetype=obj -O0 -o %t.o %s
 ; RUN: llvm-objdump -D --arch-name=amdgcn -mllvm --amdhsa-code-object-version=6 --mcpu=gfx9-generic %t.o > %t-specify.txt
 ; RUN: llvm-objdump -D  -mllvm --amdhsa-code-object-version=6 %t.o > %t-detect.txt
+;
+; RUN: llc -mtriple=amdgcn-amd-amdhsa -mcpu=gfx950 -filetype=obj -O0 -o %t.o %s
+; RUN: llvm-objdump -D --arch-name=amdgcn --mcpu=gfx950 %t.o > %t-specify.txt
+; RUN: llvm-objdump -D %t.o > %t-detect.txt
```
Comment (jayfoad) [inline llvm/test/tools/llvm-objdump/ELF/AMDGPU/subtarget.ll:152]: You've lost the "diff" run line for the clause above

File: llvm/lib/Target/AMDGPU/VOPInstructions.td [hunk 3]
```diff
@@ -650,15 +650,15 @@ class VOP_SDWA_Pseudo <string opName, VOPProfile P, list<dag> pattern=[]> :
   let Uses = !if(ReadsModeReg, [MODE, EXEC], [EXEC]);
 
   let SubtargetPredicate = HasSDWA;
-  let AssemblerPredicate = HasSDWA;
+  //let AssemblerPredicate = HasSDWA;
```
Comment (kosarev) [inline llvm/lib/Target/AMDGPU/VOPInstructions.td:653]: Just remove the line?

Comment (arsenm) [reply llvm/lib/Target/AMDGPU/VOPInstructions.td:653]: This should be in #116288

**Comment by arsenm** [comment]
### Merge activity

* **Nov 18, 1:34 PM EST**: A user started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/llvm/llvm-project/116307).

**Pull request merged**